File: Application/Imports/ServerTypings/AllocationMethod.ts
namespace Matrix {
    export enum AllocationMethod {
        ActualCost = 65,
        FifoAllocation = 70,
        LifoAllocation = 76,
        NoAllocation = 78
    }

    export class AllocationMethodOption  {
        Mapping: KeyedCollection<string>;
        constructor() {
        this.Mapping =new KeyedCollection<string>();
        this.Mapping.Add("A", "Actual Cost");   //65,
        this.Mapping.Add("F", "First-In-First-Out");   //70,
        this.Mapping.Add("L", "Last-In-First-Out");   //76,
        this.Mapping.Add("N", "No Allocation");   //78
        }
    }

    Serenity.Decorators.registerEnum(AllocationMethod, 'Enums.AllocationMethod');
}

File: Application/Modules/AdminLTE/Common/Enum/AllocationMethod.cs
#region Enterprise Solution
// Copyright © 2006-2018 Enterprise Solution Limited
// Solution:  Enterprise Solution
// Project:   Enterprise Edition/Application
// File:  AllocationMethod.cs
// Date: 2018/04/03 20:51

#endregion
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Text;
using Microsoft.Common;
using Serenity.ComponentModel;

namespace Matrix
{
    [EnumKey("Enums.AllocationMethod")]
    public enum AllocationMethod
    {
        [Description("No Allocation")]
        NoAllocation='N',

        [Description("First-In-First-Out")]
        FifoAllocation='F',

        [Description("Last-In-First-Out")]
        LifoAllocation='L',

        [Description("Actual Cost")]
        ActualCost='A'
    }
}
